packages/core/src/generators/init/generator.spec.ts

```diff
@@ -77,4 +83,27 @@ describe('init generator', () => {
       'npm run clean && npm run build && nx g @nx-dotnet/core:restore',
     );
   });
+
+  it('should add directory build props and targets files', async () => {
+    await generator(appTree, null, dotnetClient);
+    const hasPropsFile = appTree.isFile('Directory.Build.props');
+    expect(hasPropsFile).toBeTruthy();
+
+    const hasTargetsFile = appTree.isFile('Directory.Build.targets');
+    expect(hasTargetsFile).toBeTruthy();
+    const hasPreBuildTask = appTree
+      .read('Directory.Build.targets', 'utf-8')
+      ?.includes('check-module-boundaries.js');
+    expect(hasPreBuildTask).toBeTruthy();
+  });
+
+  it('should not add directory build props and targets files if props file exists', async () => {
+    appTree.write('Directory.Build.props', '');
+    const spy = jest.spyOn(devkit, 'generateFiles');
+    await generator(appTree, null, dotnetClient);
+
+    expect(spy).not.toHaveBeenCalled();
+    const hasTargetsFile = appTree.isFile('Directory.Build.targets');
+    expect(hasTargetsFile).toBeFalsy();
+  });
```

Comment:
This is good for now, in future I'd like to update these operations to add the target if it's not in there already even if file already exists.